pkg/core/pipeline/autodiscovery/config/main.go

```diff
@@ -22,8 +22,8 @@ type Input struct {
 	ScmSpec *scm.Config
 	// ScmID defines the scmid associated to the scm specification
 	ScmID string
-	// PullRequestSpecSpec defines the pullrequest specificiation
-	PullRequestSpec *pullrequest.Config
-	// ScmID defines the scmid associated to the scm specification
-	PullrequestID string
+	// ActionSpec defines the action specification
+	ActionSpec *action.Config
```

Comment:
@olblak why is it a "Spec" but points to a "Config"? I thought the convention was:
- Spec => direct unmarshalling from the user-specified YAML
- Config => internal representation of a configuration determined by a spec (such as `Init(Spec)` returns a valid Config)
?

Reply:
You are totally right. I don't remember if it's a mistake on my side, or a left over from before we decide to enforce spec

Reply:
While you are here, could you also rename it to stay consistent?